tests/cases/standalone/common/range/by.result

```diff
@@ -0,0 +1,45 @@
+CREATE TABLE host (
+  ts timestamp(3) time index,
+  host STRING PRIMARY KEY,
+  val BIGINT,
+);
+
+Affected Rows: 0
+
+INSERT INTO TABLE host VALUES
+    (0,     'host1', 0),
+    (5000,  'host1', null),
+    (10000, 'host1', 1),
+    (15000, 'host1', null),
+    (20000, 'host1', 2),
+    (0,     'host2', 3),
+    (5000,  'host2', null),
+    (10000, 'host2', 4),
+    (15000, 'host2', null),
+    (20000, 'host2', 5);
+
+Affected Rows: 10
+
+-- Test by calculate
+SELECT ts, length(host), max(val) RANGE '5s' FROM host ALIGN '20s' BY (length(host)) ORDER BY ts;
+
++---------------------+-----------------------------+----------------------------------+
+| ts                  | character_length(host.host) | MAX(host.val) RANGE 5s FILL NULL |
++---------------------+-----------------------------+----------------------------------+
+| 1970-01-01T00:00:00 | 5                           | 3                                |
+| 1970-01-01T00:00:20 | 5                           | 5                                |
++---------------------+-----------------------------+----------------------------------+
+
+SELECT ts, CAST(length(host) as INT64) + 2, max(val) RANGE '5s' FROM host ALIGN '20s' BY (CAST(length(host) as INT64) + 2) ORDER BY ts;
```

Comment:
Not sure if the `cast` here is necessary? can datafusion do this conversion automatically?

Reply:
Current version of datafusion we are currently using does not have implicit type conversion capabilities，the result of the `length` function is `int64`, `2` is `Int32`, and the two variables cannot be added directly.